ably/realtime_presence.go

```diff
@@ -25,11 +26,36 @@ type RealtimePresence struct {
 }
 
 func newRealtimePresence(channel *RealtimeChannel) *RealtimePresence {
-	return &RealtimePresence{
+	pres := &RealtimePresence{
 		subs:    newSubscriptions(subscriptionPresenceMessages),
 		channel: channel,
+		event:   make(chan State, 1),
 		members: make(map[string]*proto.PresenceMessage),
 	}
+	pres.syncMtx.Lock() // initially we lock waiting for channel's attached state
```

Comment:
I think you should avoid using short comments like this and explain it in a global comment at the top of your function.